master/internal/api_workspace.go

```diff
@@ -341,7 +341,12 @@ func (a *apiServer) PostWorkspace(
 	}
 
 	_, err = tx.NewInsert().Model(w).Exec(ctx)
+
 	if err != nil {
+		if strings.Contains(err.Error(), "violates unique constraint") {
```

Comment:
can we use pgconn.pgError.Code to detect the unique constraint error instead? it looks like the error code for a duplicate name constraint violation is "23505".

Reply:
Sounds great! I'll make that change here, as well as in the model PR 😄